tiledb/sm/stats/stats.cc

```diff
@@ -295,6 +328,20 @@ Stats* Stats::create_child(const std::string& prefix) {
 }
 
 Stats* Stats::create_child(const std::string& prefix, const StatsData& data) {
+#if !defined(TILEDB_STATS)
+  constexpr bool stats_enabled = false;
+#else
+  const bool stats_enabled = enabled_;
+#endif
+
+  if (!stats_enabled) {
+    // Return a singleton null stats object that's safe to use but does nothing.
+    // This is necessary because the caller expects a valid (non-null) pointer.
+    // Also, this avoids unnecessary allocations when stats are disabled.
+    static Stats null_stats("null_stats", false);
+    return &null_stats;
```

Comment:
Okay, so the crucial assumption here is that all of the write methods are no-ops, so this is always empty and all the reads return nothing.

This definitely raises eyebrows. I do agree with you that this should be safe but I would like to check first -

AFAIK there's two reasons for the memory bloat, right?  One is that each Stats objects builds up its counter/timer map which bloats memory, and I suppose the other reason is the accumulation of children.  Do you have some sense for how much bloat we gain just from the children?  i.e. what do we observe if the child tree structure is preserved but each one has empty data structures?

Reply:
This is from a simple write loop using TileDB-Py into a 2-dimensional dense array (both `uint64` dims) with one fixed-size `float32` attribute. Each write contains 10 elements. At every 100 writes I log RSS from `psutil`.

**Empty maps, children only** (this branch - with the original `Stats::create_child`)

* Crosses 100 MB RSS around iteration `9700`

**Full maps + children** (`main` branch)

* Crosses 100 MB RSS around iteration `3200`

**No children at all** (this branch)

* No memory accumulation

So with both counter/timer maps and children accumulating, we hit 100 MB about **3× sooner**.